lib/Parse/ParseType.cpp

```diff
@@ -722,9 +723,16 @@ Parser::parseTypeIdentifier(bool isParsingQualifiedDeclBaseType) {
       SourceLoc LAngle, RAngle;
       SmallVector<TypeRepr*, 8> GenericArgs;
       if (startsWithLess(Tok)) {
-        auto genericArgsStatus = parseGenericArguments(GenericArgs, LAngle, RAngle);
-        if (genericArgsStatus.isErrorOrHasCompletion())
-          return genericArgsStatus;
+        // Only attempt to parse a generic argument list in a cast destination
+        // type if the token text is just "<", because it can be an operator,
+        // for example: "1 as Int16 << 7".
+        if (Tok.getText().str() == "<" ||
```

Comment:
Very nit, but we can use `equals` here to compare directly and avoid `str()` 
```suggestion
        if (Tok.getText().equals("<") ||
```

Reply:
Thank you! I will use `.equals`